docs/components/DocumentationPage.tsx

```diff
@@ -50,23 +49,22 @@ const HIDDEN_ON_DESKTOP = css`
   }
 `;
 
-type Props = {
+type Props = React.PropsWithChildren<{
   router: NextRouter;
   title: string;
   sourceCodeUrl?: string;
-  /** API Page NPM package name, exposed through context for various React components that consistently use the package name. */
-  packageName?: string;
   tocVisible: boolean;
-  /* If the page should not show up in the Algolia Docsearch results */
+  /** If the page should not show up in the Algolia Docsearch results */
   hideFromSearch?: boolean;
-};
+  version: string;
+}>;
 
 type State = {
   isMenuActive: boolean;
   isMobileSearchActive: boolean;
 };
 
-export default class DocumentationPage extends React.Component<Props, State> {
+class DocumentationPageWithApiVersion extends React.Component<Props, State> {
```

Comment:
Any reason why we're leaving this as a class? Could make this all one function component with the API version hook. Totally okay if you were just preventing scope creep, just wondering if there was a reason.

Reply:
It still has a lot of old code to fully refactor this to a functional component. We still need to refactor:
- The scroll refs and the scroll layout
- Selecting the category of navigation by url
- Set header meta tags for a page
- Rendering mobile layout vs desktop
- Smooth scroll to a header by ID

That's what this component is handling, besides ofc, rendering everything 🙈